contracts/protocol/libs/MemoryUtils.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/utils/StorageSlot.sol";
+
+library MemoryUtils {
```

Comment:
Using a single core memory library with namespacing. This makes it easier to reason about every storage slot used across our contracts